Fix (S3-Backup-Setting): ensure recipients and sender are correctly fetched

AndyOverLord opened the pull request:
<!--

Some key notes before you open a PR:

 1. Select which branch should this PR be merged in?
 2. PR name follows [convention](http://karma-runner.github.io/4.0/dev/git-commit-msg.html)
 3. All tests pass locally, UI and Unit tests
 4. All business logic and validations must be on the server-side
 5. Update necessary Documentation
 6. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes


Also, if you're new here

- Documentation Guidelines => https://github.com/frappe/erpnext/wiki/Updating-Documentation

- Contribution Guide => https://github.com/frappe/frappe/blob/develop/.github/CONTRIBUTING.md

- Pull Request Checklist => https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

-->



<!-- You can skip this if you're fixing a typo or updating existing documentation -->



<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->


<!-- Add images/recordings to better visualize the change: expected/current behviour -->

> Please provide enough information so that others can review your pull request:

When setting up AWS S3 in the ERPNext, I found some issues regarding 'Email not sending after tick successfully sending to the notify_email'.

> Explain the **details** for making this change. What existing problem does the pull request solve?

Please see below links with reasons and changes I made to fix the issue:
1. [update the correct field name for S3-backup-setting](https://github.com/frappe/frappe/commit/d56742ba2948cbb71a116083e13625e49d06c3b8)
2. [sender is not set to current user](https://github.com/frappe/frappe/commit/45442728441575d9ee2287c012e950f90a3973dd)

AndyOverLord committed:
fix: sender is not set to current user

I am not sure if I did it right:
1. import frappe
2. check if the sender is None: using frappe.session.user to fetch email address for current user

AndyOverLord committed:
fix: update the correct field name for S3-backup-setting

The field variable name should be updated to 'notify_email' instead of 'notification_email'.

AndyOverLord committed:
style: remove a trailing whitespace

scmmishra commented on frappe/__init__.py:
```diff
@@ -448,7 +448,8 @@ def sendmail(recipients=[], sender="", subject="No Subject", message="No Message
 	:param header: Append header in email
 	"""
-
+	if not sender:
```
The default sender is added in `queue.py`
https://github.com/frappe/frappe/blob/1831021a77968f8770d060900849872c69108e5a/frappe/email/queue.py#L75-L77

No reason to do this here

AndyOverLord replied:
Hi Scmmishra,

Thanks for the reply. You are right about the change in ``frappe/__init__`` is not necessary. Howerver, we perhaps need to change the documentation (or comments) in line 438:
   ``:param sender: Email sender. Default is current user.`` to ``Default is outgoing email account``
   
As the codes you show at ``queue.py``, it fetches the outgoing_email_account's email address, not the current user's email address. For example, in my instance, I use support (support@...) as outgoing email account, but use administrator as user for set up S3-backup (admin@....).

But it should be all good, not a major problem. Last thing, I am not very sure how to  "port this fix to version-12-hotfix branch", could you please specify the steps for me?

Thank you,

scmmishra reviewed (commented):
The changes in `s3_backup_settings.py` is fine, however the changes in `frappe/__init__` seems unnecessary. 

Also, please port this fix to version-12-hotfix branch

Thunderbottom commented on frappe/integrations/doctype/s3_backup_settings/s3_backup_settings.py:
```diff
@@ -106,8 +106,8 @@ def send_email(success, service_name, error_status=None):
 
-	if frappe.db.get_value("S3 Backup Settings", None, "notification_email"):
-		recipients = split_emails(frappe.db.get_value("S3 Backup Settings", None, "notification_email"))
+	if frappe.db.get_value("S3 Backup Settings", None, "notify_email"):
```
Hi, while you're at this could we assign this to a variable and then check for the value in the variable instead of calling `frappe.db.get_value()` twice?

something like this should do:
```python3
	notify_email = frappe.db.get_value("S3 Backup Settings", None, "notify_email")
	if notify_email:
		recipients = split_emails(notify_email)
	...
```

AndyOverLord replied:
Good idea. I will make changes accordingly to your suggestion. 

Another thing I would like to ask, as I see the notify_email is a **mandatory** field in the form of 'S3 Backup Setting', so can I assume client-side javascript will do the validation of a correct Email address in 'notify_email' field. Can I skip the ``if notify_email:`` or should I add a ``else:  # set notify_email to the administrator's email ``?

Thank you.

Thunderbottom replied:
yeah, good catch, we can skip the if condition.

AndyOverLord committed:
fix: Assign notify_email instead of calling frappe.db.get_value twice

AndyOverLord commented:
Hi, @scmmishra 

I have write a new pull request trying to port the fix to [**version-12-hotfix.**](https://github.com/frappe/frappe/pull/9435)

But the continuous-integration/travis-ci/pr failed. I tried to look at the [job log](https://travis-ci.org/frappe/frappe/jobs/647062789?utm_medium=notification&utm_source=github_status), found traceback: 
``Traceback (most recent call last):
  File "/home/travis/frappe-bench/env/lib/python3.6/site-packages/redis/connection.py", line 484, in connect
    sock = self._connect()
  File "/home/travis/frappe-bench/env/lib/python3.6/site-packages/redis/connection.py", line 541, in _connect
    raise err
  File "/home/travis/frappe-bench/env/lib/python3.6/site-packages/redis/connection.py", line 529, in _connect
    sock.connect(socket_address)
ConnectionRefusedError: [Errno 111] Connection refused``

Not sure how to fix this, I assume it's the testing server issue?

Can you please assist with me or guides me how to solve this? Thank you.

Thunderbottom committed:
chore: remove default sender and notify_email check

* default sender already gets set in queue.py, so there's no need for a
  check inside __init__.py

* notify_email seems to be a mandatory field, so there's no need to
  check if the field has a value (assuming that it always will).

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

Thunderbottom commented:
> Hi, @scmmishra
> 
> I have write a new pull request trying to port the fix to [**version-12-hotfix.**](https://github.com/frappe/frappe/pull/9435)
> 
> But the continuous-integration/travis-ci/pr failed. I tried to look at the [job log](https://travis-ci.org/frappe/frappe/jobs/647062789?utm_medium=notification&utm_source=github_status), found traceback:
> `Traceback (most recent call last): File "/home/travis/frappe-bench/env/lib/python3.6/site-packages/redis/connection.py", line 484, in connect sock = self._connect() File "/home/travis/frappe-bench/env/lib/python3.6/site-packages/redis/connection.py", line 541, in _connect raise err File "/home/travis/frappe-bench/env/lib/python3.6/site-packages/redis/connection.py", line 529, in _connect sock.connect(socket_address) ConnectionRefusedError: [Errno 111] Connection refused`
> 
> Not sure how to fix this, I assume it's the testing server issue?
> 
> Can you please assist with me or guides me how to solve this? Thank you.


seems like an issue caused by the new werkzeug release. I've raised a PR for it here: #9437
